packages/server/src/koaJsonRpc/index.ts

```diff
@@ -106,6 +109,7 @@ export default class KoaJsonRpc {
   rpcApp(): (ctx: Koa.Context, _next: Koa.Next) => Promise<void> {
     return async (ctx: Koa.Context, _next: Koa.Next) => {
       this.requestId = ctx.state.reqId;
+      this.requestIpAddress = ctx.request.ip;
```

Comment:
Q: should we only set this if the relay operator plans to rate limit by IP?

Reply:
@Nana-EC  the idea here is that we need this for the hbar limiter and spending plan, so we always want to have the sender's ip in order to check if its a part of the spending plan of a partner. so its not related to rate limiting but hbar limiting and we need it passed to sdkClient and the functions there

Reply:
My point is if IP based hbar limiting isn't enabled then this should be skipped.

Reply:
If we want to disable IP-based limiting we can do it inside the `HbarLimitService` for better locality of behavior.

This PR which wraps the `requestId` into a `requestDetails` object should be agnostic of the hbar limiter. This is more of an overall change for the whole project. We can still have access to the IP address, but only conditionally use it inside `HbarLimitService`.